app/js/mms/controllers/view.controller.js

```diff
@@ -188,6 +189,29 @@ angular.module('mmsApp')
         $scope.bbApi.toggleButtonSpinner('center-next');
     });
 
+    // Share URL button settings
+    $scope.dynamicPopover = {
+        templateUrl: 'shareUrlTemplate.html',
+        title: 'Share'
+    };
+
+    $scope.copyToClipboard = function ($event) {
+        $event.stopPropagation();
+        var target = $('#ve-short-url');
+        UtilsService.copyToClipboard(target);
+    };
+
+    $scope.handleShareURL = function() {
+        var currentUrl = URLService.getMmsServer() + $location.url();
```

Comment:
shouldn't it just use absUrl() so it's exactly what's in the location bar? localhost is fine since it'll just be for developers testing anywyas